src/PowerShell/Microsoft.WinGet.Client/Helpers/ComObjectFactory.cs

```diff
@@ -87,18 +101,46 @@ public virtual UninstallOptions CreateUninstallOptions()
         /// <returns>A <see cref="PackageMatchFilter" /> instance.</returns>
         public virtual PackageMatchFilter CreatePackageMatchFilter()
         {
-            return Create<PackageMatchFilter>(PackageMatchFilterType);
+            return Create<PackageMatchFilter>(PackageMatchFilterType, PackageMatchFilterIid);
         }
 
-        private static T Create<T>(Type type)
+        private static T Create<T>(Type type, in Guid iid)
         {
-            object instance = Activator.CreateInstance(type);
+            object instance;
+
+            if (Utilities.ExecutingAsAdministrator)
+            {
+                instance = null;
+
+                int hr = WinGetServerManualActivation_CreateInstance(type.GUID, iid, 0, out instance);
+
+                if (hr == ErrorCode.ERROR_FILE_NOT_FOUND)
+                {
+                    throw new Exception(Utilities.ResourceManager.GetString("WinGetPackageNotInstalled"));
+                }
+                else if (hr != 0)

```

Comment:
>hr != 0 [](http://example.com/codeflow?start=25&length=7)

This should use same logic as SUCCEEDED(hr)
 that is used in WinGetServerManualActivation_CreateInstance implementation